src/api/app/controllers/webui/project_controller.rb

```diff
@@ -455,15 +455,8 @@ def monitor
       arch: @arch_filter, repo: @repo_filter }
     find_opt[:lastbuild] = 1 unless @lastbuild_switch.blank?
 
-    @buildresult = Buildresult.find( find_opt )
-    unless @buildresult
-      flash[:warning] = "No build results for project '#{@project}'"
-      redirect_to action: :show, project: params[:project]
-      return
-    end
-
-    @buildresult = @buildresult.to_hash
-    unless @buildresult.has_key? 'result'
+    @buildresult = Buildresult.find( find_opt ).try { to_hash }
```

Comment:
`try` is not needed, `find` raise an `ActiveRecord::RecordNotFound` error if the object cannot be found, so it can not be `null`

http://api.rubyonrails.org/classes/ActiveRecord/Associations/CollectionProxy.html#method-i-find